### ISSUE TITLE
How to get Agent Execution running, no output from server

### ISSUE THREAD

**Issue by unoriginalscreenname** [paolorechia/learn-langchain#28]
I'm having an issue where I'm trying to run an example using a zero shot agent and a basic tool via your short_instruction example.  

If I load in the OpenAI api as the LLM and run all the other code in the example I get exactly what I'd expect printed out in the console:
> Entering new AgentExecutor chain...
 I need to create a csv file and save the jokes to it.
Action: Python REPL
Action Input:
import csv

with open('catjokes.csv', 'w', newline='') as csvfile:
    writer = csv.writer(csvfile, delimiter=',')
    writer.writerow(['Joke 1', 'Joke 2', 'Joke 3'])
    writer.writerow(['Why did the cat go to the party?', 'Because he was feline fine!', 'Why did the cat cross the road? To get to the meowtel!'])
Observation:
Thought: I now know how to save cat jokes to a csv file.

However, if I switch to the Vicuna server and run everything, with the only difference being the llm is now coming from the local server, I get nothing back in the console and my GPU gets stuck processing something but I can't tell what's going on.  

Are you able to run these examples locally?  I have this feeling like there's some piece of information being left out here.  All agent based examples running locally through here exhibit the same behavior.  It must be something to do with what's being passed into the model server, but I can't figure it out.

Thoughts?

**Comment by paolorechia** [comment]
Hi, that’s strange, as the examples work for me. 

If you’re executing the Vicuna server, you should be able to see its logs - is it receiving the request normally but then get stuck at the processing?

Did you try reducing the number of max new tokens parameter to see if it’s just being really slow?

Also which model are you using, is it quantized?

**Comment by paolorechia** [comment]
Almost of the current examples were tested using the WizardLM 7b uploaded by the bloke. Maybe it’s worth trying that one instead - stable vicuna didn’t work well for me if I recall correctly

**Comment by paolorechia** [comment]
However it’s strange that’s so slow, shouldn’t be - it was only this slow when I tried using the beams parameter or a bigger model like starcoder

**Comment by paolorechia** [comment]
Also something which is not great: I observed quantized models perform worse in these tasks - which is why I stick with the HF format these days

**Comment by paolorechia** [comment]
Can’t say all, but most of them worked for me last time I tested.

Also it could be that something with the quantized implementation in my
server is buggy - it wouldn’t surprise me as I’ve stopped using this
implementation ever since I started supporting oobagooba’s server

I still use the vicuna server with the hf format though, was running it
today in fact to run some experiments, and it was generating output normally



TestUser ***@***.***> schrieb am Do. 11. Mai 2023 um 21:18:

**Comment by paolorechia** [comment]
Yeah, depends on the model / query. Models that don’t perform well as an agent tend to return invalid responses all the time
